cmd/main.go

```diff
@@ -3,13 +3,21 @@ package main
 import "thing"
 
 func main() {
+
+	packInstall := make([]string, 5)
```

Comment:
We can refactor this a little now based on our recent design thinking, keeping in mind the principle that we should squash as much domain-specific knowledge as possible into our package, relieving the user of the need to know it!

For example, we want to install a gem, and we _know_ that the `ruby` and `gem` packages are prerequisites for that. So let's let the _user_ express what they want to do in the minimal code:

```go
r.InstallGem("jekyll")
r.InstallGem("bundler")
```

and we can then implement that inside our package as something like:

```go
func InstallGem(p string) {
	r.UpdateYum() // for efficiency, make sure we only do this once per program invocation—how?
	if !r.IsInstalled("ruby") {
		r.Install("ruby")
	}
	if !r.IsInstalled("gem") {
		r.Install("gem")
	}
	if !r.IsInstalledGem(p) {
		err := r.Command("gem", "install "+p)
		if err != nil {
			return err
		}
	}
}
```

...you get the idea.

Reply:
Ah, I think I'm following, to the user s/he does not need to know or check if ruby or gems are installed. Their goal is to install jekyll, so why not allow them to focus on the front-end packages, right? Do they need to know ruby and ... are required, no. 

Okay, I'll try implementing the different runner objects first then move on to this section. Thanks!